test/e2e/bcd_consumer_integration/clientcontroller/babylon/babylon.go

```diff
@@ -0,0 +1,538 @@
+	pagination := &sdkquery.PageRequest{
+		Limit:   count,
+		Reverse: reverse,
+		Key:     startKey,
+	}
+
+	res, err := bc.bbnClient.QueryClient.ListBlocks(status, pagination)
+	if err != nil {
+		return nil, fmt.Errorf("failed to query finalized blocks: %v", err)
+	}
+
+	for _, b := range res.Blocks {
+		ib := &types2.BlockInfo{
+			Height: b.Height,
+			Hash:   b.AppHash,
+		}
+		blocks = append(blocks, ib)
+	}
+
+	return blocks, nil
+}
+
+func (bc *BabylonController) Close() error {
+	if !bc.bbnClient.IsRunning() {
+		return nil
+	}
+
+	return bc.bbnClient.Stop()
+}
+
+/*
+	Implementations for e2e tests only
+*/
+
+func (bc *BabylonController) CreateBTCDelegation(
+	delBtcPk *bbntypes.BIP340PubKey,
+	fpPks []*btcec.PublicKey,
+	pop *btcstakingtypes.ProofOfPossessionBTC,
+	stakingTime uint32,
+	stakingValue int64,
+	stakingTxInfo *btcctypes.TransactionInfo,
+	slashingTx *btcstakingtypes.BTCSlashingTx,
+	delSlashingSig *bbntypes.BIP340Signature,
+	unbondingTx []byte,
+	unbondingTime uint32,
+	unbondingValue int64,
+	unbondingSlashingTx *btcstakingtypes.BTCSlashingTx,
+	delUnbondingSlashingSig *bbntypes.BIP340Signature,
+) (*types2.TxResponse, error) {
+	fpBtcPks := make([]bbntypes.BIP340PubKey, 0, len(fpPks))
+	for _, v := range fpPks {
+		fpBtcPks = append(fpBtcPks, *bbntypes.NewBIP340PubKeyFromBTCPK(v))
+	}
+	msg := &btcstakingtypes.MsgCreateBTCDelegation{
+		StakerAddr:                    bc.MustGetTxSigner(),
+		Pop:                           pop,
+		BtcPk:                         delBtcPk,
+		FpBtcPkList:                   fpBtcPks,
+		StakingTime:                   stakingTime,
+		StakingValue:                  stakingValue,
+		StakingTx:                     stakingTxInfo,
+		SlashingTx:                    slashingTx,
+		DelegatorSlashingSig:          delSlashingSig,
+		UnbondingTx:                   unbondingTx,
+		UnbondingTime:                 unbondingTime,
+		UnbondingValue:                unbondingValue,
+		UnbondingSlashingTx:           unbondingSlashingTx,
+		DelegatorUnbondingSlashingSig: delUnbondingSlashingSig,
+	}
+
+	res, err := bc.reliablySendMsg(msg, emptyErrs, emptyErrs)
+	if err != nil {
+		return nil, err
+	}
+
+	return &types2.TxResponse{TxHash: res.TxHash}, nil
+}
+
+func (bc *BabylonController) InsertBtcBlockHeaders(headers []bbntypes.BTCHeaderBytes) (*provider.RelayerTxResponse, error) {
+	msg := &btclctypes.MsgInsertHeaders{
+		Signer:  bc.MustGetTxSigner(),
+		Headers: headers,
+	}
+
+	res, err := bc.reliablySendMsg(msg, emptyErrs, emptyErrs)
+	if err != nil {
+		return nil, err
+	}
+
+	return res, nil
+}
+
+// TODO: only used in test. this should not be put here. it causes confusion that this is a method
+// that will be used when FP runs. in that's the case, it implies it should work all all consumer
+// types. but `bbnClient.QueryClient.FinalityProviders` doesn't work for consumer chains
+func (bc *BabylonController) QueryFinalityProviders() ([]*btcstakingtypes.FinalityProviderResponse, error) {
```

Comment:
This looks like a duplicated query. Why not use the one in `configurer/chain/queries_btcstaking.go` by example? The same potentially with the other queries below.